integration/spark/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -0,0 +1,173 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.hive
+
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.CarbonTableIdentifierImplicit
+import org.apache.spark.sql.catalyst.analysis.{UnresolvedAlias, UnresolvedFunction, UnresolvedRelation, UnresolvedStar}
+import org.apache.spark.sql.catalyst.expressions.Alias
+import org.apache.spark.sql.catalyst.plans.Inner
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules._
+import org.apache.spark.sql.execution.command.ProjectForDeleteCommand
+import org.apache.spark.sql.execution.datasources.LogicalRelation
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+
+/**
+ * Insert into carbon table from other source
+ */
+object CarbonPreInsertionCasts extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.transform {
+      // Wait until children are resolved.
+      case p: LogicalPlan if !p.childrenResolved => p
+
+      case p @ InsertIntoTable(relation: LogicalRelation, _, child, _, _)
+           if relation.relation.isInstanceOf[CarbonDatasourceRelation] =>
+        castChildOutput(p, relation.relation.asInstanceOf[CarbonDatasourceRelation], child)
+    }
+
+    def castChildOutput(p: InsertIntoTable, relation: CarbonDatasourceRelation, child: LogicalPlan)
+      : LogicalPlan = {
+      if (relation.carbonRelation.output.size > CarbonCommonConstants
+        .DEFAULT_MAX_NUMBER_OF_COLUMNS) {
+        sys
+          .error("Maximum supported column by carbon is:" + CarbonCommonConstants
+            .DEFAULT_MAX_NUMBER_OF_COLUMNS
+          )
+      }
+      if (child.output.size >= relation.carbonRelation.output.size ) {
+        InsertIntoCarbonTable(relation, p.partition, p.child, p.overwrite, p.ifNotExists)
+      } else {
+         sys.error("Cannot insert into target table because column number are different")
+      }
+    }
+  }
+
+object CarbonIUDAnalysisRule extends Rule[LogicalPlan] {
+
+  var sqlContext: SQLContext = _
+
+  def init(sqlContext: SQLContext) {
+    this.sqlContext = sqlContext
+  }
+
+  private def processUpdateQuery(
+                   table: UnresolvedRelation,
+                   columns: List[String],
+                   selectStmt: String,
+                   filter: String): LogicalPlan = {
+    var includedDestColumns = false
+    var includedDestRelation = false
+    var addedTupleId = false
+
+    def prepareTargetReleation(relation: UnresolvedRelation): Subquery = {
+      val tupleId = UnresolvedAlias(Alias(UnresolvedFunction("getTupleId",
+        Seq.empty, isDistinct = false), "tupleId")())
+      val projList = Seq(
+        UnresolvedAlias(UnresolvedStar(table.alias)), tupleId)
```

Comment:
@ravikiran23  @jackylk UnresolvedStar's init method requests a ‘Option[String]’ type in Spark-1.5 while in Spark-1.6 it requests a 'Option[Seq[String]]' type.